state-storage/lib/state-storage.ts

```diff
@@ -32,14 +46,15 @@ export class StateStorage {
      * @param {string} service name of the service (class name?) that owns the state
      * @param {string} name name of the state variable to save
      * @param value content to save
-     * @param quiet if true, don't log content
-     * @returns {Promise<void>} completes upon success - failures shouldn't happen
+     * @param optionsOrQuiet a StateStorageOptions, or if true sets quiet option. (For backward compatibility.)
+     * @returns {Promise<void>} completes upon success - rejects if lacking ssm:PutParameter permission
      */
-    set(service: string, name: string, value: any, quiet = false): Promise<void> {
+    set(service: string, name: string, value: any, optionsOrQuiet: boolean | StateStorageOptions = {}): Promise<void> {
+        const options = optionsOrQuiet === true ? {quiet: true} : optionsOrQuiet as StateStorageOptions;
         const key = this.generateName(service, name);
-        const content = JSON.stringify(value);
+        const content = options.isRaw === true ? value : JSON.stringify(value);
```

Comment:
rather than using ternary on individual props, you could set a default options object and use the spread operator to combine user defined and default properties.